pytorch_lightning/trainer/connectors/precision_connector.py

```diff
@@ -54,9 +59,15 @@ def _setup_amp_backend(self, amp_type: str):
                                ' We will attempt to use NVIDIA Apex for this session.')
                 amp_type = 'apex'
             else:
-                log.info('Using native 16bit precision.')
                 self.trainer.amp_backend = AMPType.NATIVE
-                self.backend = NativeAMPPlugin(self.trainer)
+                if plugins and self._sharded_in_plugins(plugins):
+                    if not FAIRSCALE_AVAILABLE:
+                        raise MisconfigurationException('Sharded DDP Plugin requires Fairscale to be installed.')
+                    log.info('Using Sharded 16bit plugin.')
+                    self.backend = ShardedNativeAMPPlugin(self.trainer)
+                else:
+                    log.info('Using native 16bit precision.')
+                    self.backend = NativeAMPPlugin(self.trainer)
 
         if amp_type == 'apex':
```

Comment:
Does it work with Apex ?

Reply:
at this point, it is just string...

Reply:
It doesn't, and it isn't supported by fairscale yet since apex has the same issue with scaling gradients. Might be worth adding a check to ensure the user doesn't use both at the same time!